tensorflow/lite/kernels/batch_matmul_test.cc

```diff
@@ -279,6 +279,67 @@ INSTANTIATE_TEST_SUITE_P(
     BatchMatMulOpTest, BatchMatMulOpTest,
     ::testing::ValuesIn(SingleOpTest::GetKernelTags(*kKernelMap)));
 
+class ConstRHSBatchMatMulOpModel : public MultiOpModel {
+ public:
+  ConstRHSBatchMatMulOpModel(const TensorData& lhs,
+                             std::initializer_list<int> rhs_shape,
+                             std::initializer_list<float> rhs_data,
+                             bool adj_x = false, bool adj_y = false) {
+    lhs_id_ = AddInput(lhs);
+    rhs_id_ = AddConstInput<float>(TensorType_FLOAT32, rhs_data, rhs_shape);
+    matmul_output_id_ = AddOutput(lhs.type);
+    std::vector<int> matmul_inputs{lhs_id_, rhs_id_};
+    std::vector<int> matmul_outputs{matmul_output_id_};
+    AddBuiltinOp(BuiltinOperator_BATCH_MATMUL,
+                 BuiltinOptions_BatchMatMulOptions,
+                 CreateBatchMatMulOptions(builder_, adj_x, adj_y).Union(),
+                 matmul_inputs, matmul_outputs);
+
+    // This is just a dummy op to check if the values of transposed RHS are
```

Comment:
Can you explain why the neg op help here?

Reply:
It's somehow tricky. When there is no following ops (not limited to neg), temporary allocation with `kTfLiteArenaRw` type tends to get the same piece of memory across each evaluation and no other ops will modify values at that memory address (cause no other memory allocation). Therefore, the values of rhs are not polluted without neg op, and the next evaluation can also get the same piece of memory, so the results are correct.

https://colab.research.google.com/drive/1eFVszZZN-NcR64X_VHns0w-l3Vwci0CY?usp=sharing

Here is the colab link from original post. The values the last tensor (temp tensor, transposed rhs) become

```
[[-51. -84.]
 [  3.   6.]
 [  7.   9.]]
```

where -51, -84 are the last two elements of `Identity`. While without neg op, everything works smoothly.

Reply:
Acked. Thanks for explanation.  May be it worths a comment then.

Reply:
Updated comments. See if it's better! Thanks!